server_modules/scores.js

```diff
@@ -47,4 +71,46 @@ exports.route = function(app) {
         }).catch(utils.sendError(res)).done();
     });
 
+    //save a new score
+    app.post('/scores/create',function(req,res) {
+        var body = JSON.parse(req.body);
+        var scoresheet = body.scoresheet;
+        var score = body.score;
+
+        changeScores(function(result) {
+            result.scores.push(score);
+            return result;
+        })
+        .then(fileSystem.writeFile(fileSystem.getDataFilePath("scoresheets/" + score.file), req.body))
```

Comment:
This is the wrong order of changing the files: what happens if the `scores.json` was written, but scoresheet can't be saved?

Also remember that writing the scoresheet is not a problem if it happens multiple times (i.e. is idempotent).